The default progname really should be kpsewhich, and makes a test
pass which otherwise fails.

--- texk/kpathsea/kpsewhich.c	2020-07-01 19:07:23.592000000 +1000
+++ texk/kpathsea/kpsewhich.c	2020-07-01 19:07:33.999000000 +1000
@@ -60,7 +60,7 @@
 boolean must_exist = false;
 
 /* The program name, for `.PROG' construct in texmf.cnf.  (-program) */
-string progname = NULL;
+string progname = "kpsewhich";
 
 /* Safe input and output names to check.  (-safe-in-name and -safe-out-name) */
 string safe_in_name = NULL;
